db/queries/seller.sql

```diff
@@ -0,0 +1,416 @@
+    c."expire_date"
+FROM "coupon" c
+    JOIN "shop" s ON c."shop_id" = s.id
+WHERE s.seller_name = $1
+ORDER BY "start_date" DESC
+LIMIT $2
+OFFSET $3;
+
+-- name: SellerGetCouponDetail :one
+
+SELECT c.*
+FROM "coupon" c
+    JOIN "shop" s ON c."shop_id" = s.id
+WHERE
+    s."seller_name" = $1
+    AND c."id" = $2;
+
+-- name: SellerInsertCoupon :one
+
+INSERT INTO
+    "coupon" (
+        "type",
+        "shop_id",
+        "name",
+        "description",
+        "discount",
+        "start_date",
+        "expire_date"
+    )
+VALUES (
+        $2, (
+            SELECT s."id"
+            FROM "shop" s
+            WHERE
+                s."seller_name" = $1
+                AND s."enabled" = true
+        ),
+        $3,
+        $4,
+        $5,
+        $6,
+        $7
+    )
+RETURNING *;
+
+-- name: UpdateCouponInfo :one
+
+UPDATE "coupon" c
+SET
+    "type" = COALESCE($3, "type"),
+    "name" = COALESCE($4, "name"),
+    "description" = COALESCE($5, "description"),
+    "discount" = COALESCE($6, "discount"),
+    "start_date" = COALESCE($7, "start_date"),
+    "expire_date" = COALESCE($8, "expire_date")
+WHERE c."id" = $2 AND "shop_id" = (
+        SELECT s."id"
+        FROM "shop" s
+        WHERE
+            s."seller_name" = $1
+            AND s."enabled" = true
+    )
+RETURNING *;
+
+-- name: SellerDeleteCoupon :execrows
+
+DELETE FROM "coupon" c
+WHERE c."id" = $2 AND "shop_id" = (
+        SELECT s."id"
+        FROM "shop" s
+        WHERE
+            s."seller_name" = $1
+            AND s."enabled" = true
+    );
+
+-- name: SellerGetOrder :one
+
+SELECT
+    "id",
+    "shipment",
+    "total_price",
+    "status",
+    "created_at"
+FROM "order_history"
+WHERE "shop_id" = (
+        SELECT s."id"
+        FROM "shop" s
+        WHERE
+            s."seller_name" = $1
+            AND s."enabled" = true
+    )
+ORDER BY "created_at" DESC
+LIMIT $2
+OFFSET $3;
+
+-- name: SellerOrderCheck :one
+
+SELECT order_history.*
+FROM "order_history"
+    JOIN shop ON order_history.shop_id = shop.id
+WHERE
+    shop.seller_name = $1
+    AND order_history.id = $2;
+
+-- name: SellerGetOrderDetail :many
+
+SELECT
+    product_archive.*,
+    order_detail.quantity
+FROM "order_detail"
+    LEFT JOIN product_archive ON order_detail.product_id = product_archive.id AND order_detail.product_version = product_archive.version
+    LEFT JOIN order_history ON order_history.id = order_detail.order_id
+    LEFT JOIN shop ON order_history.shop_id = shop.id
+WHERE
+    shop.seller_name = $1
+    AND order_detail.order_id = $2
+ORDER BY quantity * price DESC
+LIMIT $3
+OFFSET $4;
+
+-- name: SellerUpdateOrderStatus :one
+
+UPDATE "order_history" oh
+SET
+    "status" = sqlc.arg(set_status)
+WHERE "shop_id" = (
+        SELECT s."id"
+        FROM "shop" s
+        WHERE
+            s."seller_name" = $1
+            AND s."enabled" = true
+    )
+    AND oh."id" = $2
+    AND oh."status" = sqlc.arg(current_status)
+RETURNING *;
+
```

Comment:
why not just use $n instead of sqlc.arg

Reply:
Then it will let the second status variable be named 'status_2' or something generated by sqlc